### PR TITLE
Context cards migration

### PR DESCRIPTION
**Pull request by Tehsurfer**
**Updates 08/09**
*Flatmap context card moved to here and sidebar context card disabled*

This PR  moves the context card from being on the sidebar to on mapintagratedvuer in a popper js popper.

The same inputs are passed from the sidebar as previously used.

It stores a list of context cards and checks the slot id and name in splitpaneheader.vue for visibility.

It requires small updates to map-sidebar and scaffoldvuer to work (note that they are updated in package.json)

This PR also includes a small change to pass facet labels to scaffolvuer, as previously only facet uberons were passed in. This is is used to update markers on the scaffold.

Context cards on split pane:
image

Markers updating with algolia active filters:
image

image

### CODE REVIEW

**Comment by Tehsurfer** [comment]
hmm not sure why this is failing the tests, the video looks ok

File: src/components/ContextCard.vue [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+<script>
+/* eslint-disable no-alert, no-console */
+import Vue from "vue";
+import { Link, Icon, Card, Button, Select, Input } from "element-ui";
+import lang from "element-ui/lib/locale/lang/en";
+import locale from "element-ui/lib/locale";
+import EventBus from "./EventBus"
+
+//provide the s3Bucket related methods and data.
+import S3Bucket from "../mixins/S3Bucket";
```
Comment (alan-wu) [inline src/components/ContextCard.vue:74]: This file is not included in previous commit.

Comment (Tehsurfer) [reply src/components/ContextCard.vue:74]: oops sorry! I have added it now

File: src/components/SplitpanesBar.vue [hunk 2]
```diff
@@ -34,6 +34,35 @@
...
+        <div v-show="contextCardVisible[slot.name]" class="hide" @click="contextCardVisible[slot.name] = false">
+          Hide information
+          <i class="el-icon-arrow-up"></i>
+        </div>
+        <div v-show="!contextCardVisible[slot.name]" class="hide" @click="contextCardVisible[slot.name] = true">
+          Show information
+          <i class="el-icon-arrow-down"></i>
+        </div>
+        <el-popover
+          v-if="container != undefined"
```
Comment (alan-wu) [inline src/components/SplitpanesBar.vue:48]: This line should be removed.

**Review by alan-wu — COMMENTED** [review]
Can the show-information button be hidden on panel without contextual information?

File: src/components/SplitpanesBar.vue [hunk 3]
```diff
@@ -100,6 +139,29 @@ export default {
...
+        ALGOLIA_ID: store.state.settings.algoliaId,
+        PENNSIEVE_API_LOCATION: store.state.settings.pennsieveApi,
+        NL_LINK_PREFIX: store.state.settings.nlLinkPrefix,
+        ROOT_URL: store.state.settings.rootUrl,
+      };
+    },
+    boundary: function () {
+      let b = document.querySelector(".tab-container")
+      return b;
+    },
```
Comment (alan-wu) [inline src/components/SplitpanesBar.vue:157]: The tab-container class is now named container after the change get reverted.

By the way, it is probably better to add a @hook:mounted method in the SplitpaneBar components of the template in splitDialog which will then pass in the boundary into it.

**Review by alan-wu — CHANGES_REQUESTED** [review]
Probably better to not pass mapImp as a prop.

File: src/components/viewers/MultiFlatmap.vue [hunk 4]
```diff
@@ -234,6 +234,7 @@ export default {
         this.flatmapReady = true;
         const flatmapImp = flatmap.mapImp;
         this.flatmapMarkerZoomUpdate(true, flatmapImp);
+        EventBus.$emit("mapImp", flatmapImp); // send mapimp to the context card for provenance
```
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply src/components/viewers/MultiFlatmap.vue:237]: oh true good point

File: src/components/SplitpanesBar.vue [hunk 5]
```diff
@@ -34,6 +34,46 @@
...
+          class="context-card-popover"
+          :popper-options="popperOptions"
+          v-model="contextCardVisible[slot.name]"
+        >
+          <template v-for="(contextCardEntry, i) in contextCardEntries">
+              <flatmap-context-card 
+                class="flatmap-context-card"
+                :key="'flatmapContextCard'+i" 
+                v-if="contextCardEntry.id === slot.id && contextCardEntry.type == 'flatmap'" 
+                :mapImp="contextCardEntry.mapImp"
```
[the remark this answers is not in the document]
Comment (Tehsurfer) [reply src/components/SplitpanesBar.vue:62]: yeah sure! Good idea

**Review by alan-wu — APPROVED** [review]
It looks good and works well now.

**Pull request merged**